src/transformers/models/vivit/configuration_vivit.py

```diff
@@ -0,0 +1,119 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Vivit model configuration"""
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+VIVIT_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "jegormeister/vivit-b-16x2-kinetics400": (
+        "https://huggingface.co/jegormeister/vivit-b-16x2-kinetics400/resolve/main/config.json"
+    ),
+    # See all Vivit models at https://huggingface.co/models?filter=vivit
+}
+
+
+class VivitConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`VivitModel`]. It is used to instantiate a Vivit
+    model according to the specified arguments, defining the model architecture. Instantiating a configuration with the
+    defaults will yield a similar configuration to that of the Vivit
+    [jegormeister/vivit-b-16x2-kinetics400](https://huggingface.co/jegormeister/vivit-b-16x2-kinetics400) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+
+    Args:
+        video_size (`List[int]`, *optional*, defaults to [32, 224, 224]):
+            The size (resolution) of a video in the format `[number_of_frames, width, height]`.
+        tubelet_size (`List[int]`, *optional*, defaults to [2, 16, 16]):
+            The size (resolution) of each tubelet.
```

Comment:
(nit) for consistency with other models like ViT and VideoMAE, it'd be great to have a `num_frames` attribute in the config, and an `image_size` attribute (rather than video_size). The `image_size` can be an integer or a tuple.

See also my comment in the `TubeletEmbeddings` class